lib/IRGen/DebugTypeInfo.h

```diff
@@ -116,7 +117,10 @@ class DebugTypeInfo {
   llvm::Optional<Size::int_type> getRawSizeInBits() const { return SizeInBits; }
   Alignment getAlignment() const { return Align; }
   bool isNull() const { return Type == nullptr; }
-  bool isForwardDecl() const { return FragmentStorageType == nullptr; }
+  bool isForwardDecl() const {
+    return FragmentStorageType == nullptr &&
+           !llvm::isa<swift::GenericTypeParamType>(Type);
```

Comment:
Ah I see. So only generic types are not forward declarations?

Reply:
Yes, but on second thought, maybe keeping generic type parameters types as forward declarations is better after all. I'll undo these changes.